### ISSUE TITLE
Compression type matching and automatic decompression

### ISSUE THREAD

**Issue by treythomas123** [h2non/gock#11]
It'd be nice to have a way to specify a compression type, for testing clients that gzip their body contents. For example:

```go
gock.New(url).
    Post("/path").
    MatchCompression("gzip").
    JSON(map[string]interface{}{
        "foo": "bar",
    }).
    Reply(200)
```

To accomplish this now requires a custom matcher, since the `JSON` matcher assumes the request body is unencoded.

I can write a PR if you're interested in including this.

Thanks!

**Comment by h2non** [comment]
That would be great!  Using just `Compression()` will keep it shorter but equally expressive.

**Comment by h2non** [comment]
This feature is now shipped in `gock@1.0.2`. Thanks!
